lib/AxiosCall.js

```diff
@@ -0,0 +1,77 @@
+const axios = require('axios').default;
+
+class AxiosCall {
+
+    constructor(method, server, params, callbackSuccess, callbackError){
+        this.config = {
+            headers: {
+              'Content-Type': 'application/json'
+            }
+        }
+        this.callbackSuccess = callbackSuccess;
+        this.callbackError = callbackError;
+        this.server = server;
+        this.body = {
+            method: method || '',
+            params: params
+        };
+    }
+
+    getCallUrl(method){
+        let server = this.server || 'my.geotab.com';
+        let thisServer = server.replace(/\S*:\/\//, '').replace(/\/$/, '');
+        return 'https://' + thisServer + '/apiv1' + (method ? '/' + method : '');
+    }  
+
+    encodeRPC(data){
+        let rpcString = JSON.stringify({
+            method: data.method || '',
+            params: data.params
+        })
+        let rpcData = `JSON-RPC=${encodeURIComponent(rpcString)}`;
+        return rpcData;
+    }
+
+    /**
+     * Sends axios request
+     * @param {int} timeout amount in milliseconds for timeout. 
+     *                      options.timeout * 1000 is a good start
+     *                      Defaults to no timeout (0)
+     * @returns {Promise} Axios promise
+     */
+    async send(timeout = 0){
+        this.request = axios({
+            method: 'POST',
+            url: this.getCallUrl(this.body.method),
+            data: this.encodeRPC(this.body),
+            config: this.config,
```

Comment:
I noticed that request was still being made as `content-type: ` and this is why you were required to still add `JSON-RPC=${encodeURIComponent(rpcString)}` to `encodeRpc` method. I tried changing this to `headers: this.config.headers` and modified `encodeRpc` to just return `rpcString`. Which seemed to work.

> In result of axios I could see config property showing header sent